x/shared/types/query_client.go

```diff
@@ -0,0 +1,34 @@
+package types
+
+import (
+	"context"
+
+	gogogrpc "github.com/cosmos/gogoproto/grpc"
+)
+
+// SharedQueryClient is an interface which adapts generated (concrete) shared query client
+// to paramsQuerierIface (see: pkg/client/query/paramsquerier.go) such that implementors
+// (i.e. the generated shared query client) is compliant with client.ParamsQuerier for the
+// shared module's params type. This is required to resolve generic type constraints.
+type SharedQueryClient interface {
+	QueryClient
+	GetParams(context.Context) (*Params, error)
```

Comment:
```suggestion
	GetParams(context.Context) (*Params, error)
	// TODO_MAINNET(@bryanchriswhite, #931): GetParamsAtHeight(context.Context, uint64) (*Params, error)
```

Reply:
This is not accurate. The point of this interface is only to adapt from:

```go
Params(context.Context) *Params
```

to:
```go
GetParams(context.Context) (*Params, error)
```

This interface is used as a generic type parameter to the `NewCachedParamsQuerier()` constructor which is what returns the object that defines `GetParamsAtHeight().`

Reply:
This is not accurate. The point of this interface is only to adapt from:

```go
Params(context.Context, *QueryParamsRequest, ...) (*QueryParamsResponse, error)
```

to:
```go
GetParams(context.Context) (*Params, error)
```

This interface is used as a generic type parameter to the `NewCachedParamsQuerier()` constructor which is what returns the object that defines `GetParamsAtHeight().`

Reply:
Can you #PUC so this reasoning is clear to others please?